app/controllers/application_controller/compare.rb

```diff
@@ -412,14 +412,60 @@ def sections_field_changed
     end
   end
 
-  def set_checked_sections
-    session[:selected_sections] = []
-    params[:all_checked]&.each do |a|
-      s = a.split(':')
-      if s.length > 1
-        session[:selected_sections].push(s[1])
+  # Method to get the selection names from :all_checked and :id params.
+  # individual selected item suggests that the variable name contains ':'
+  # else, the child_names are retrived from the selected_parent item.
+  def selection_names(name)
```

Comment:
All of these new helper methods should be made private if they are not accessed from a route.